server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java

```diff
@@ -0,0 +1,112 @@
+
+   @ClassRule public static InfinispanServerRule SERVERS = ClusteredIT.SERVERS;
+
+   @Rule public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+   @Test public void testContinuousQuery() throws IOException {
+      ConfigurationBuilder config = new ConfigurationBuilder();
+      config.marshaller(new ProtoStreamMarshaller());
+      RemoteCache<String, Person> cache = SERVER_TEST.getHotRodCache(config, CacheMode.DIST_SYNC);
+      RemoteCacheManager rcm = cache.getRemoteCacheManager();
+
+      SerializationContext serializationContext = MarshallerUtil.getSerializationContext(rcm);
+      ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
+      String protoFile = protoSchemaBuilder.fileName("test.proto").addClass(Person.class).build(serializationContext);
+
+      RemoteCache<String, String> metadataCache = rcm
+            .getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
+      metadataCache.put("test.proto", protoFile);
+      assertFalse(metadataCache.containsKey(ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX));
+      cache.clear();
+      QueryFactory qf = Search.getQueryFactory(cache);
+      Query query = qf.from(Person.class).having("name").eq("Adrian").build();
+
+      ContinuousQuery<String, Person> continuousQuery = Search.getContinuousQuery(cache);
+
+      BlockingQueue<String> people = new LinkedBlockingQueue<>();
+      ContinuousQueryListener<String, Person> listener = new ContinuousQueryListener<String, Person>() {
+         @Override public void resultJoining(String key, Person value) {
+            people.add(value.name);
+         }
+      };
+      continuousQuery.addContinuousQueryListener(query, listener);
+      cache.put("Adrian", new Person("Adrian", 1));
+      assertTrue(cache.containsKey("Adrian"));
+      expectElementsInQueue(people, 1);
+   }
+
+   private void assertTrue(boolean adrian) {
+   }
+
+   private <T, R> void expectElementsInQueue(BlockingQueue<T> queue, int numElements) {
+      for (int i = 0; i < numElements; i++) {
+         final T o;
+         try {
+            o = queue.poll(5, TimeUnit.SECONDS);
+            assertNotNull("Queue was empty after reading " + i + " elements!", o);
+         } catch (InterruptedException e) {
+            throw new AssertionError("Interrupted while waiting for condition", e);
+         }
+      }
+
+      try {
+         // no more elements expected here
+         Object o = queue.poll(5, TimeUnit.SECONDS);
+         assertNull("No more elements expected in queue!", o);
+      } catch (InterruptedException e) {
+         throw new AssertionError("Interrupted while waiting for condition", e);
+      }
+   }
+
+   public static class Person {
+
+      @ProtoField(number = 1) public String name;
+
+      @ProtoField(number = 2) public Integer id;
+
+      public Person() {
+      }
+
+      public Person(String name, Integer id) {
```

Comment:
FYI you can do now the following with the latest protostream release:

```java
      @ProtoField(number = 1)
      public String name;

      @ProtoField(number = 2)
      public Integer id;

      @ProtoFactory
      public Person(String name, Integer id) {
```

So there's no need for the no-arg constructor

Reply:
btw @ryanemerson these new marshalling works with quarkus ?

Reply:
wondering for here: https://github.com/infinispan-demos/harry-potter-quarkus/pull/8

Reply:
To answer simply: Yes

Reply:
https://github.com/quarkusio/quarkus/blob/master/integration-tests/infinispan-client/src/main/java/io/quarkus/it/infinispan/client/Author.java is an example